### ISSUE TITLE
Version list is not populated when running behind nginx

### ISSUE THREAD

**Issue by morhekil** [ArekSredzki/electron-release-server#96]
I have the server running behind nginx, that is configured to proxy web socket connections. The problem is that in this configuration application front-end doesn't seem to be rendering the list of versions correctly. I can create a version, and I can see that the version is written into the database, but the list of versions is still coming up empty all the time.

Looking at the network traffic and comparing it to direct connection (e.g. a server running locally) - it looks like after a ws connection is established, the application doesn't receive from the server the ws frame containing the list of available versions.

There appears to be a similar issue #89 - but I'd like to reopen the problem, as that one didn't provide a real solution.

I'm happy to debug it further, but I would appreciate any hints/suggestions, as I'm not that familiar with sails.js or angular to diagnose the problem myself.

**Comment by ArekSredzki** [comment]
Hi @morhekil sorry for the delay.
This seems like an issue in setting up nginx.
Did you follow the instructions [here](https://github.com/ArekSredzki/electron-release-server/blob/master/docs/deploy.md) on adding the appropriate proxy headers and upgrading the connection?

**Comment by morhekil** [comment]
Hi @ArekSredzki,

yes, nginx is set up for HTTP connection upgrades, and I can see websocket connection established in Chrome's dev tools, and some empty frames being received by the browser. The problem is that the actual data frames are not coming down, and thus nothing is displayed in the UI.

**Comment by ArekSredzki** [comment]
Interesting.
What are the versions of the following:
- Server OS
- Nginx
- Node
- Chrome

Also, are you sure that nothing else is involved? Ex. Using Cloudflare

**Comment by ArekSredzki** [comment]
Also, do you have a firewall?

**Comment by morhekil** [comment]
Ubuntu 16.04.1 LTS, running its standard Nginx 1.10.0, and node 7.2.0 from deb.nodesource.com. Chrome is at v55 at the moment. The whole system is running on Google Compute Engine, protected by GCE's firewalls (but with http/https traffic allowed through to nginx).

Interesting thing is that I recently tried to open a connection directly to the release server itself, bypassing nginx altogether, and observed the same problem happening in this setup. So it might be that GCE it doing something weird with its networking that messes up release server's communications, and not nginx itself.

**Comment by ArekSredzki** [comment]
Oh interesting.
I haven't tested node 7.x, and recommend you use 6.x. Try that and let me know if anything changes :)

**Comment by rgl** [comment]
Versions are not displayed here too. I'm using a similar setup, but with node 6.x and running in a VirtualBox VM.

**Comment by rgl** [comment]
Oh it works fine. I didn't restart nginx after setting the nginx settings to be like the ones described at [deploy.md](https://github.com/ArekSredzki/electron-release-server/blob/master/docs/deploy.md). Sorry for the noise!

**Comment by ArekSredzki** [comment]
@morhekil Did you have any luck with Node 6.9.1?

**Comment by morhekil** [comment]
sorry guys, just got flooded with other work this week. I'll try to test it with Node 6 over the weekend, and if still doesn't work - rerun the same server playbook at another cloud provider (Digital Ocean?) to trying and pinpoint the source of the problem, and report back with the results.

**Comment by ArekSredzki** [comment]
Great :) Unfortunately you can't use docker on Digital Ocean (nested virtualization is a nightmare), but the server should work perfectly in a normal setup.

**Comment by rgl** [comment]
@ArekSredzki Docker does not use virtualization. You should be able to use Docker on DigitalOcean.

**Comment by ArekSredzki** [comment]
@rgl My bad. Since it uses virtualization on MacOS and Windows, I assumed that it did on linux as well. Very cool 👌

**Comment by ionphractal** [comment]
TL;DR I just ran into a similar (maybe the same) problem except that I'm using Caddy as reverse proxy. In my case I was missing WebSocket configuration in the proxy. Seems like the Sails doesn't fall back to polling and insists of opening a WebSocket connection.
@morhekil could you open the web / JS dev console in your browser and see if there are any WebSocket related connection errors? If so you have the culprit. Take a look at https://www.nginx.com/blog/websocket-nginx/ on how to configure it (basically just need to add http1.1 and set the appropriate upgrade header).
